maintainers/scripts/pluginupdate.py

```diff
@@ -425,27 +443,101 @@ def generate_nix(self, _plugins, _outfile: str):
+            of spaghetti code, which I don't want to write.
+
+        Arguments:
+            plugin: Plugin on which you decide whether to ignore or not.
+            to_update:
+                List of strings, that user passed to `--update` parameter.
+                By default, we assume it is a list of URIs identical to what
+                is in input file.
+
+        Returns:
+            True if we should update plugin and False if not.
+        """
+        return plugin.name.replace(".", "-") in to_update
+
+    def get_update(self, input_file: str, outfile: str, config: FetchConfig, to_update: Optional[List[str]]):
+        if to_update is None:
+            to_update = []
+
+        current_plugins = self.get_current_plugins(config, self.nixpkgs)
+        current_plugin_specs = self.load_plugin_spec(config, input_file)
+
+        cache: Cache = Cache(map(lambda x: x[1], current_plugins), self.cache_file)
         _prefetch = functools.partial(prefetch, cache=cache)
 
-        def update() -> dict:
-            plugins = self.load_plugin_spec(config, input_file)
+        filter_func = functools.partial(self.filter_plugins_to_update, to_update=to_update)
+        plugins_to_update = list(
+            filter(filter_func, current_plugin_specs)
+            if to_update
+            else current_plugin_specs
+        )
+
+        def update() -> Redirects:
+            if len(plugins_to_update) == 0:
+                log.error(
+                    "\n\n\n\nIt seems like you provided some arguments to `--update`:\n"
+                    + ", ".join(to_update) +
+                    "\nBut after filtering, the result list of plugins is empty\n"
+                    "\n"
+                    "Are you sure you provided the same URIs as in your input file?\n"
+                    "(" + str(input_file) + ")\n\n"
+                )
+                return {}
 
             try:
                 pool = Pool(processes=config.proc)
-                results = pool.map(_prefetch, plugins)
+                results = pool.map(_prefetch, plugins_to_update)
             finally:
                 cache.store()
 
+            print(f"{len(results)} of {len(current_plugins)} were checked")
+            # Do only partial update of out file
+            if len(results) != len(current_plugins):
+                results = self.merge_results(current_plugins, results)
             plugins, redirects = check_results(results)
 
+            plugins = sorted(plugins, key=lambda v: v[1].normalized_name)
             self.generate_nix(plugins, outfile)
 
             return redirects
 
         return update
 
+    def merge_results(
+        self,
+        current: list[Tuple[PluginDesc, Plugin]],
+        fetched: List[Tuple[PluginDesc, Union[Exception, Plugin], Optional[Repo]]],
+    ) -> List[Tuple[PluginDesc, Union[Exception, Plugin], Optional[Repo]]]:
+        result: Dict[str, Tuple[PluginDesc, Union[Exception, Plugin], Optional[Repo]]] = dict(
```

Comment:
Those types are copied from `check_results` func, and result of this function will be passed to it